apps/web/lib/trpc/routers/keySettings.ts

```diff
@@ -0,0 +1,409 @@
+          description: `updated key ${input.keyId} rate limit`,
+          keyId: input.keyId,
+        });
+      });
+    }),
+  updateOwnerId: t.procedure
+    .use(auth)
+    .input(
+      z.object({
+        keyId: z.string(),
+        ownerId: z.string().nullish(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const key = await db.query.keys.findFirst({
+        where: (table, { eq }) => eq(table.id, input.keyId),
+        with: {
+          workspace: true,
+        },
+      });
+      if (!key) {
+        throw new TRPCError({ message: "key not found", code: "NOT_FOUND" });
+      }
+      if (key.workspace.tenantId !== ctx.tenant.id) {
+        throw new TRPCError({ message: "key not found", code: "NOT_FOUND" });
+      }
+      await db.transaction(async (tx) => {
+        await tx
+          .update(schema.keys)
+          .set({
+            ownerId: input.ownerId ?? null,
+          })
+          .where(eq(schema.keys.id, input.keyId));
+        await tx.insert(schema.auditLogs).values({
+          id: newId("auditLog"),
+          time: new Date(),
+          workspaceId: key.workspaceId,
+          apiId: null,
+          actorType: "user",
+          actorId: ctx.user.id,
+          event: "key.update",
+          description: `updated key ${input.keyId} ownerId`,
+          keyId: input.keyId,
+        });
+      });
+
+      return true;
+    }),
+  updateName: t.procedure
+    .use(auth)
+    .input(
+      z.object({
+        keyId: z.string(),
+        name: z.string().nullish(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const key = await db.query.keys.findFirst({
+        where: (table, { eq }) => eq(table.id, input.keyId),
+        with: {
+          workspace: true,
+        },
+      });
+      if (!key) {
+        throw new TRPCError({ message: "key not found", code: "NOT_FOUND" });
+      }
+      if (key.workspace.tenantId !== ctx.tenant.id) {
+        throw new TRPCError({ message: "key not found", code: "NOT_FOUND" });
+      }
+      await db.transaction(async (tx) => {
+        await tx
+          .update(schema.keys)
+          .set({
+            name: input.name ?? null,
+          })
+          .where(eq(schema.keys.id, input.keyId));
+        await tx.insert(schema.auditLogs).values({
+          id: newId("auditLog"),
+          time: new Date(),
+          workspaceId: key.workspaceId,
+          apiId: null,
+          actorType: "user",
+          actorId: ctx.user.id,
+          event: "key.update",
+          description: `updated key ${input.keyId} name`,
+          keyId: input.keyId,
+        });
+      });
+      return true;
+    }),
+  updateMetadata: t.procedure
+    .use(auth)
+    .input(
+      z.object({
+        keyId: z.string(),
+        metadata: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      let meta: unknown | null = null;
+
+      if (input.metadata === null || input.metadata === "") {
+        meta = "";
```

Comment:
might as well set it to the correct value already
```suggestion
        meta = null;
```

Reply:
Changed